Test/Tests/RealtimeClientChannelTests.swift

```diff
@@ -521,15 +522,12 @@ class RealtimeClientChannelTests: XCTestCase {
                     fail("Error is nil"); done(); return
                 }
                 XCTAssertEqual(error.code, ARTErrorCode.tokenExpired.intValue)
-
-                channel.on { stateChange in
-                    if stateChange.current == .attached {
-                        XCTAssertTrue(stateChange.resumed)
-                        XCTAssertNil(stateChange.reason)
-                        XCTAssertEqual(stateChange.current, ARTRealtimeChannelState.attached)
-                        XCTAssertEqual(stateChange.previous, ARTRealtimeChannelState.attached)
-                        done()
-                    }
+                XCTAssertEqual(channel.state, ARTRealtimeChannelState.attached)
```

Comment:
I don't understand the idea behind these test changes — why is it no longer making assertions about the `ATTACHED` `ChannelStateChange`, which is what RTL2f talks about? And how is it related to the above change, which relates to when an `UPDATE` `ChannelStateChange` will be emitted?

Reply:
Looks like RTL2f talks about `ChannelStateChange` missing details of `RTL12` which states that only in case of `resumed == false` `channel.on` will be triggered. This test checks the case when `resumed == true`, that's why I've removed `channel.on` (bc it will not be triggered). So the new assertions claim that channel states remains the same before and after ATTACHED protocol message and that it contains `resumed == true`.

Reply:
`RTL2f` is about what value to use for the `resumed` value of a `ChannelStateChange`. That means that this test is no longer testing `RTL2f` as it advertises, since as you've said there is no channel state change any more.

Reply:
I've [removed](https://github.com/ably/ably-cocoa/pull/1791/commits/1a5f933f4068398009cf0641bfdfb76844fa5baa) `RTL2f` test and created [issue](https://github.com/ably/ably-cocoa/issues/1812) to create a proper one.

Reply:
Also updated `RTL12` test - https://github.com/ably/ably-cocoa/pull/1791/commits/9184efa3eece85f60db12d0cf69cada8a6ba5319